src/tls13.c

```diff
@@ -0,0 +1,5343 @@
+    XMEMCPY(&data[idx], info, infoLen);
+    idx += infoLen;
+
+    WOLFSSL_MSG("PRK");
+    WOLFSSL_BUFFER(prk, prkLen);
+    WOLFSSL_MSG("Info");
+    WOLFSSL_BUFFER(data, idx);
+
+    ret = wc_HKDF_Expand(digest, prk, prkLen, data, idx, okm, okmLen);
+
+    WOLFSSL_MSG("OKM");
+    WOLFSSL_BUFFER(okm, okmLen);
+
+    ForceZero(data, idx);
+
+    return ret;
+}
+
+/* Size of the TLS v1.3 label use when deriving keys. */
+#define TLS13_PROTOCOL_LABEL_SZ    9
+/* The protocol label for TLS v1.3. */
+static const byte tls13ProtocolLabel[TLS13_PROTOCOL_LABEL_SZ + 1] = "TLS 1.3, ";
+
+/* Derive a key from a message.
+ *
+ * ssl        The SSL/TLS object.
+ * output     The buffer to hold the derived key.
+ * outputLen  The length of the derived key.
+ * secret     The secret used to derive the key (HMAC secret).
+ * label      The label used to distinguish the context.
+ * labelLen   The length of the label.
+ * msg        The message data to derive key from.
+ * msgLen     The length of the message data to derive key from.
+ * hashAlgo   The hash algorithm to use in the HMAC.
+ * returns 0 on success, otherwise failure.
+ */
+static int DeriveKeyMsg(WOLFSSL* ssl, byte* output, int outputLen,
+                        const byte* secret, const byte* label, word32 labelLen,
+                        byte* msg, int msgLen, int hashAlgo)
+{
+    byte        hash[MAX_DIGEST_SIZE];
+    Digest      digest;
+    word32      hashSz = 0;
+    const byte* protocol;
+    word32      protocolLen;
+    int         digestAlg;
+
+    switch (hashAlgo) {
+#ifndef NO_WOLFSSL_SHA256
+        case sha256_mac:
+            wc_InitSha256(&digest.sha256);
+            wc_Sha256Update(&digest.sha256, msg, msgLen);
+            wc_Sha256Final(&digest.sha256, hash);
+            wc_Sha256Free(&digest.sha256);
+            hashSz = SHA256_DIGEST_SIZE;
+            digestAlg = SHA256;
+            break;
+#endif
+#ifdef WOLFSSL_SHA384
+        case sha384_mac:
+            wc_InitSha384(&digest.sha384);
+            wc_Sha384Update(&digest.sha384, msg, msgLen);
+            wc_Sha384Final(&digest.sha384, hash);
+            wc_Sha384Free(&digest.sha384);
+            hashSz = SHA384_DIGEST_SIZE;
+            digestAlg = SHA384;
+            break;
+#endif
+#ifdef WOLFSSL_SHA512
+        case sha512_mac:
+            wc_InitSha512(&digest.sha512);
+            wc_Sha512Update(&digest.sha512, msg, msgLen);
+            wc_Sha512Final(&digest.sha512, hash);
+            wc_Sha512Free(&digest.sha512);
+            hashSz = SHA512_DIGEST_SIZE;
+            digestAlg = SHA512;
+            break;
+#endif
+
+        default:
+            return BAD_FUNC_ARG;
+    }
+
+    switch (ssl->version.minor) {
+        case TLSv1_3_MINOR:
+            protocol = tls13ProtocolLabel;
+            protocolLen = TLS13_PROTOCOL_LABEL_SZ;
+            break;
+    }
+    if (outputLen == -1)
+        outputLen = hashSz;
+
+    return HKDF_Expand_Label(output, outputLen, secret, hashSz,
```

Comment:
clang scan-build is reporting that _protocol_ and _protocolLen_ are being used without being initialized.